### PR TITLE
Fixing incumbent

### PR DESCRIPTION
**Pull request by LindsayYoung**
there are some holes in this data so we want the lat time it was updated

The example we have been using for testing is Nancy Pelosi

### CODE REVIEW

File: data/sql_updates/create_candidate_detail_view.sql [hunk 1]
```diff
@@ -54,6 +54,13 @@ from dimcand
         select distinct on (cand_sk) * from dimcandproperties
             order by cand_sk, candproperties_sk desc
     ) cand_p_most_recent using(cand_sk, cand_id)
+    -- there are some holes in this data so we want the lat time it was updated
+    left join(
+        select distinct on (cand_sk, cand_ici_desc, cand_ici_desc)
+            cand_sk, cand_ici_desc, cand_ici_desc
+        where cand_ici_desc != Null
```
Comment (jmcarp) [inline data/sql_updates/create_candidate_detail_view.sql]: Should this be `...is not null`?

File: data/sql_updates/create_candidate_detail_view.sql [hunk 2]
```diff
@@ -54,6 +54,13 @@ from dimcand
         select distinct on (cand_sk) * from dimcandproperties
             order by cand_sk, candproperties_sk desc
     ) cand_p_most_recent using(cand_sk, cand_id)
+    -- there are some holes in this data so we want the lat time it was updated
+    left join(
+        select distinct on (cand_sk, cand_ici_desc, cand_ici_desc)
```
Comment (jmcarp) [inline data/sql_updates/create_candidate_detail_view.sql]: `cand_ici_desc` is repeated here.

File: data/sql_updates/create_candidate_detail_view.sql [hunk 3]
```diff
@@ -54,6 +54,13 @@ from dimcand
...
+            cand_sk, cand_ici_desc, cand_ici_desc
```
Comment (jmcarp) [inline data/sql_updates/create_candidate_detail_view.sql]: `cand_ici_desc` repeated.

File: data/sql_updates/create_candidate_detail_view.sql [hunk 4]
```diff
@@ -54,6 +54,13 @@ from dimcand
...
+        where cand_ici_desc != Null
+        order by  cand_sk, cand_id, election_yr desc
+    ) full_ici using (cand_sk, cand_id)
```
Comment (jmcarp) [inline data/sql_updates/create_candidate_detail_view.sql]: Just curious, but why `...using (cand_sk, cand_id)`? Is `...using (cand_sk)` not enough?

File: data/sql_updates/create_candidate_detail_view.sql [hunk 5]
```diff
@@ -54,6 +54,13 @@ from dimcand
         select distinct on (cand_sk) * from dimcandproperties
             order by cand_sk, candproperties_sk desc
     ) cand_p_most_recent using(cand_sk, cand_id)
+    -- there are some holes in this data so we want the lat time it was updated
+    left join(
+        select distinct on (cand_sk, cand_ici_desc, cand_ici_desc)
+            cand_sk, cand_ici_desc, cand_ici_desc
+        where cand_ici_desc is not null
```
Comment (arowla) [inline data/sql_updates/create_candidate_detail_view.sql]: Missing `from` table

File: data/sql_updates/create_candidate_detail_view.sql (hunk 2 above)
Comment (arowla) [inline data/sql_updates/create_candidate_detail_view.sql]: Shouldn't this just be distinct on cand_sk?

File: data/sql_updates/create_candidate_detail_view.sql [hunk 6]
```diff
@@ -54,6 +54,14 @@ from dimcand
         select distinct on (cand_sk) * from dimcandproperties
             order by cand_sk, candproperties_sk desc
     ) cand_p_most_recent using(cand_sk, cand_id)
+    -- there are some holes in this data so we want the lat time it was updated
+    left join(
+        select distinct on (cand_sk)
+            s.cand_sk, s.cand_ici_desc, s.cand_ici_cd
```
Comment (arowla) [inline data/sql_updates/create_candidate_detail_view.sql]: Where are these columns from? `cand_ici_desc` and `cand_ici_cd` are not valid on `dimcandstatusici`

File: data/sql_updates/create_candidate_detail_view.sql [hunk 7]
```diff
@@ -54,6 +54,14 @@ from dimcand
...
+        where cand_ici_desc is not null
+        from dimcandstatusici s
```
Comment (arowla) [inline data/sql_updates/create_candidate_detail_view.sql]: Clauses out of order. FROM before WHERE.

File: data/sql_updates/create_candidate_detail_view.sql (hunk 6 above)
Comment (LindsayYoung) [inline data/sql_updates/create_candidate_detail_view.sql]: on a related note. I used to use the following to have a test data set locally:

psql -f data/cfdm_test.pgdump.sql cfdm_test

Did that change or do we not have test data anymore?

**Comment by LindsayYoung** [comment]
You know what they say, the 27th time's a charm?

**Comment by jmcarp** [comment]
LGTM!

**Pull request merged**